R/combine_mixed_crops.R

```diff
@@ -0,0 +1,123 @@
+#' Combine two-PFT outputs to represent mixed cropping systems
+#'
+#' Rules for combining woody and annual PFT outputs to represent a
+#' mixed cropping system. Supports two methods:
+#'  - "weighted": PFTs partition area (woody_cover + annual_cover = 1) and the
+#'    output is a weighted average: `woody_cover * woody_value + annual_cover * annual_value`.
+#'  - "incremental": preserves the full-area woody baseline (requires `woody_cover == 1`)
+#'    and treats annual as an increment relative to an annual initial baseline:
+#'    `annual_delta = annual_value - annual_init`; `result = woody_value + annual_cover * annual_delta`.
+#'
+#' Vectorization & recycling are handled via `vctrs::vec_recycle_common()`. All scalar
+#' inputs are broadcast to the common length. Length mismatches other than 1 or the
+#' common size trigger a severe error.
+#'
+#' Validation rules (severe errors unless otherwise noted):
+#'  * No input values may be NA (including covers, pool sizes, annual_init if required)
+#'  * Covers must lie within [0,1]
+#'  * Method "incremental": `woody_cover` must be 1 (within 0.1%); if not, severe error
+#'  * Method "incremental": `annual_init` required
+#'  * Method "weighted": rows whose `woody_cover + annual_cover` differ from 1 by more than tolerance
+#'    are set to NA in the result; a single aggregated warning is emitted listing the count
+#'
+#' @param woody_value numeric. Pool size for the woody PFT (kg/m2).
+#' @param annual_value numeric. Pool size for the annual PFT (kg/m2).
+#' @param annual_init numeric, required for method = "incremental"; the initial annual pool.
+#' @param annual_cover numeric. Fractional cover of the annual PFT (f_annual, 0-1).
+#' @param woody_cover numeric. Fractional cover of the woody PFT (f_woody, 0-1). Must be 1 for incremental.
```

Comment:
I don't see `f_annual` or `f_woody` used anywhere else user-facing -- seems like they could be deleted here without loss of clarity
```suggestion
#' @param annual_cover numeric. Fractional cover of the annual PFT (0-1).
#' @param woody_cover numeric. Fractional cover of the woody PFT (0-1). Must be 1 when `method` is "incremental".
```